assets/js/amp-block-validation.js

```diff
@@ -0,0 +1,185 @@
+					let result;
+
+					result = [];
+					originalEdit = wp.element.createElement( OriginalBlockEdit, this.props );
+					if ( originalEdit ) {
+						result.push( originalEdit );
+					}
+					if ( this.state.isInvalidAMP && wp.components.hasOwnProperty( 'Notice' ) ) {
+						result.push( wp.element.createElement(
+							wp.components.Notice,
+							{
+								status: 'warning',
+								content: module.data.i18n.notice.replace( '%s', this.props.name ),
+								isDismissible: false
+							}
+						) );
+					}
+
+					this.props.attributes.pendingValidation = false;
+					return result;
+				}
+
+				/**
+				 * Handler for after the component mounting.
+				 *
+				 * If validateAMP() changes the isInvalidAMP state, it will result in this method being called again.
+				 * There's no need to check if the state is valid again.
+				 * So this skips the check if pendingValidation is true.
+				 *
+				 * @returns {void}
+				 */
+				componentDidMount() {
+					if ( ! this.props.attributes.pendingValidation ) {
+						let content = this.props.attributes.content;
+						if ( 'string' !== typeof content ) {
+							content = wp.element.renderToString( content );
+						}
+						if ( content.length > 0 ) {
+							this.validateAMP( this.props.attributes.content );
+						}
+					}
+					this.props.attributes.pendingValidation = false;
+				}
+
+				/**
+				 * Validates the content for AMP compliance, and sets the state of the Notice.
+				 *
+				 * Depending on the results of the validation,
+				 * it sets the Notice component's isInvalidAMP state.
+				 * This will cause the notice to either display or be hidden.
+				 *
+				 * @param {string} content The block content, from calling save().
+				 * @returns {void}
+				 */
+				validateAMP( content ) {
+					this.setState( function() {
+
+								// Changing the state can cause componentDidMount() to be called, so prevent it from calling validateAMP() again.
+								component.props.attributes.pendingValidation = true;
+								return { isInvalidAMP: ( Math.random() > 0.5 ) };
+							} );
+
+					let component = this;
+					$.post(
+						module.data.endpoint,
+						{
+							markup: content
+						}
+					).done( function( data ) {
+						if ( data.hasOwnProperty( 'removed_elements' ) && ( 0 === data.removed_elements.length ) && ( 0 === data.removed_attributes.length ) ) {
+							component.setState( function() {
+
+								// Changing the state can cause componentDidMount() to be called, so prevent it from calling validateAMP() again.
+								component.props.attributes.pendingValidation = true;
+								return { isInvalidAMP: false };
+							} );
+						} else {
+							component.setState( function() {
+								component.props.attributes.pendingValidation = true;
+								return { isInvalidAMP: true };
+							} );
+						}
+					} );
+				}
+
+			};
+
+			wp.blocks.registerBlockType( blockType, block );
+		}
+
+	};
+
+	return module;
+
+} )( window.jQuery );
```

Comment:
We should eliminate jQuery as being a dependency for this script. Extending Gutenberg should be jQuery-free.

Reply:
Thanks, @westonruter. [This commit](https://github.com/Automattic/amp-wp/pull/1019/commits/ea1c37af19209b6851b261cb115105a5b5367bb4) removes the `jQuery` dependency.